pkg/virt-handler/vm.go

```diff
@@ -312,9 +320,11 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 				// Update IP info based on information from domain.Status.Interfaces (Qemu guest)
 				// Remove the interface from domainInterfaceStatusByMac to mark it as handled
 				if interfaceStatus, exists := domainInterfaceStatusByMac[interfaceMAC]; exists {
-					newInterface.IP = interfaceStatus.Ip
-					newInterface.IPs = interfaceStatus.IPs
-					newInterface.InterfaceName = interfaceStatus.InterfaceName
+					if !IsMasqueredeInterface {
```

Comment:
@vatsalparekh @phoracek Maybe it worth to put a comment before the condition to explain the possible race condition?